doozer/doozerlib/backend/konflux_client.py

```diff
@@ -0,0 +1,507 @@
+        obj["metadata"]["labels"]["appstudio.openshift.io/application"] = application_name
+        obj["metadata"]["labels"]["appstudio.openshift.io/component"] = component_name
+
+        def _modify_param(params: List, name: str, value: Union[str, bool, list[str]]):
+            """ Modify a parameter in the params list. If the parameter does not exist, it is added.
+
+            :param params: The list of parameters.
+            :param name: The name of the parameter.
+            :param value: The value of the parameter.
+            """
+            if isinstance(value, bool):
+                value = "true" if value else "false"  # boolean value should be converted to string
+            for param in params:
+                if param["name"] == name:
+                    param["value"] = value
+                    return
+            params.append({"name": name, "value": value})
+
+        # PipelineRun parameters to override in the template
+        params = obj["spec"]["params"]
+        _modify_param(params, "output-image", output_image)
+        _modify_param(params, "skip-checks", skip_checks)
+        _modify_param(params, "image-expires-after", "6w")
+        _modify_param(params, "build-platforms", list(build_platforms))
+
+        # See https://konflux-ci.dev/docs/how-tos/configuring/customizing-the-build/#configuring-timeouts
+        obj["spec"]["timeouts"] = {"pipeline": "12h"}
+
+        # Task specific parameters to override in the template
+        for task in obj["spec"]["pipelineSpec"]["tasks"]:
+            match task["name"]:
+                case "build-images":
+                    task["timeout"] = "12h"
+                case "apply-tags":
+                    _modify_param(task["params"], "ADDITIONAL_TAGS", list(additional_tags))
+
+        # https://konflux.pages.redhat.com/docs/users/how-tos/configuring/overriding-compute-resources.html
+        # ose-installer-artifacts fails with OOM with default values, hence bumping memory limit
+        obj["spec"]["taskRunSpecs"] = [{
+            "pipelineTaskName": "build-images",
+            "stepSpecs": [{
+                "name": "sbom-syft-generate",
+                "computeResources": {
+                    "requests": {
+                        "memory": "5Gi"
+                    },
+                    "limits": {
+                        "memory": "10Gi"
+                    }
+                }
+            }]
+        }]
+
+        return obj
+
+    async def start_pipeline_run_for_image_build(
+        self,
+        generate_name: str,
+        namespace: Optional[str],
+        application_name: str,
+        component_name: str,
+        git_url: str,
+        commit_sha: str,
+        target_branch: str,
+        output_image: str,
+        building_arches: Sequence[str],
+        git_auth_secret: str = "pipelines-as-code-secret",
+        additional_tags: Sequence[str] = [],
```

Comment:
default value for `additional_tags` sis mutable. In Python, mutable args are evaluated only once at function definition time, so calling the function with a different value of the argument will affect all subsequent calls of that function. Better use `None` instead

Reply:
Done